lxd/endpoints/cluster.go

```diff
@@ -45,7 +45,12 @@ func (e *Endpoints) ClusterUpdateAddress(address string) error {
 	e.closeListener(cluster)
 
 	// If turning off listening, we're done
-	if address == "" || util.IsAddressCovered(address, networkAddress) {
+	if address == "" {
+		return nil
+	}
+
+	// If the address is covered by a non-empty network address, don't do anything.
+	if networkAddress != "" && util.IsAddressCovered(address, networkAddress) {
```

Comment:
@masnax should this empty networkAddress be part of `IsAddressCovered` perhaps?

Reply:
I wasn't too sure about what we want to allow/disallow when it comes to `IsAddressCovered`, since it seems we are deliberately checking if `address2` (or `networkAddress` in this case) is an empty string in that function, and then considering the address covered. In this case we don't want an empty string, and I was hesitant to change the behaviour of that function.

Reply:
I think the `IsAddressCovered` naming could be clearer, my understanding is that its for answering the question: "will this cause a socket bind conflict" - which makes sense why if the networkAddress is empty, then it will use a wild card which will cause a conflict.

But I am unclear why that does not apply here.
Perhaps updating the comment "If the address is covered by a non-empty network address, don't do anything." to explain the *why* rather than just the *what* would help :)

Reply:
Ah, I actually did a bit of digging and we never actually proceed with opening a listener if we get an empty string for `core.https_address`, since it's also the unsetting value. I suppose that this means the empty string case can just be dropped from `IsAddressCovered`.

Reply:
Yeah sounds good, adding some tests for these scenarios would be good too, in case its regressed in the future.